spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -790,6 +806,7 @@ private void registerHandlers() {
 				if (name != null) {
 					handlersByName.put(name, monitor);
 				}
+				AbstractMessageHandlerMetrics metrics = this.metricsFactory.createHandlerMetrics(name);
```

Comment:
Adding `Assert.notNull` here too.
I really can imagine some bad guy, who aims to humiliate our code :smile: with his custom `MetricsFactory` which returns `null` :smile: